sources/packages/backend/apps/api/src/utilities/class-validation/custom-validators/json-max-size.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ValidatorConstraintInterface,
+  ValidatorConstraint,
+  registerDecorator,
+  ValidationArguments,
+} from "class-validator";
+
+/**
+ * Max size validation for the json resulted from the annotated object.
+ */
+@ValidatorConstraint()
+class JsonMaxSizeConstraint implements ValidatorConstraintInterface {
+  validate(json: unknown, args: ValidationArguments): boolean {
+    const [maxSize] = args.constraints;
+    const stringifiedJson = JSON.stringify(json);
```

Comment:
One last-minute question, if the `json` parameter is `undefined`, does it not cause an exception when `stringifiedJson.length` is checked?

Reply:
Yes,  good catch. Added a check. Thanks!